### ISSUE TITLE
Suggestion: Vehicles should be assigned to routes, not the other way around

### ISSUE THREAD

**Issue by n473-** [voxeltycoon/suggestions#251]
Routes to Vehicles is a one-to-many ratio. A route can have many vehicles, but a vehicle can only have one route. Currently, routes (along with cargo selection) are assigned to vehicles. That is to say that the primary means of managing route assignment is through a list of vehicles, and the only way to assign cargo filters is at the per-vehicle level. 

It would feel more sane, in my opinion, to create and manage routes independently, and then assign vehicles to them. One would create a route, defining each stop along with respective cargo filters, and then assign vehicles to it. Each vehicle assigned to the route would inherit the cargo configuration of the route. 

The current (simplified) flow is:

1. Buy vehicle
2. Configure vehicle's cargo filter
3. Create an individual route for the vehicle and optionally save it as a standard route
4. Optionally copy the vehicle in the garage/depot multiple times
5. Manage the route through the vehicles screen

Modifying an existing route requires clicking on one of the vehicles with that route and editing it within the vehicle window itself. Modifying existing vehicle configuration (cargo filters, etc.) requires each vehicle to be selected and sent to the garage/depot individually.

My suggested flow would be:

1. Create new route
2. Assign stops and cargo filters (possibly even vehicle types?)
3. Buy N vehicles and assign to route
4. Manage the route through a route screen

Modifying an existing route is done through the route overview window (like the existing vehicles window, but for routes). This would also enable easier bulk modifications, like upgrading/replacing/selling vehicles _per route_.

This would probably require major changes to the underlying game logic (I'm no game developer so I have no idea), so a suggestion to improve this without requiring massive changes would be this:

Make the cargo filter system part of the defined route, not part of the vehicle configuration. When you define each stop, give the option to define which specific cargo must be loaded/unloaded. Vehicles added to a route inherit the cargo filters defined in the route.

**Comment by klunkean** [comment]
I agree with you on the point that there should be an extra route overview screen. As you said, for real convenience the route configuration window is just a couple of clicks too far away.

However, you can add/change cargo filters on the route by using the "Refit" order. Works like a charm.

**Comment by n473-** [comment]
I'm familiar with the "refit" order for mid-route cargo changes, which I assume one might use if one wants to ship a load somewhere and return with a different type of load. But purely from a management perspective, if I want to switch, say, 10 trucks from an iron ore route to an iron bar route, I should be able to do that without having to individually send each truck to the garage, where I then have to manually change the filter on each one. I also shouldn't require a Refit order on every 2-stop route just to ensure my trucks are loading the correct cargo. 

I'm happy to abuse the Refit order for now, but it would be nice in future if I didn't have to constantly micromanage the cargo filters for each vehicle. Thanks for the suggestion!

**Comment by klunkean** [comment]
>  if I want to switch, say, 10 trucks from an iron ore route to an iron bar route, I should be able to do that without having to individually send each truck to the garage

But with the refit order you are able to do that without having to individually send each truck to the garage! Why is the refit order abused in this case? It is exactly doing what its name suggests.

Not having to send trucks to the garage for a filter change is planned:
> With the next update, we’ll introduce the next changes to help newcomers:
> 
> Storage will be not chosen by default, you’ll have to choose either Auto or exact cargo type explicitly (as it already working with storages)
> You’ll be able to manually refit a vehicle by clicking on the cargo icon in the Details tab of the Vehicle window, without the need to send a vehicle to a depot

**Comment by n473-** [comment]
Okay let me rephrase. I understand that there are ways to do what I would like. I haven't used the refit orders, and will do so at your suggestion. But I still feel as if this solution is unintuitive and somewhat of a hack. Refit orders, traditionally, are used when a vehicle transports more than one cargo type per route. It feels inelegant to have a vehicle go through a refit task every time it completes a route. It is part of the route specifically because the assumption is that you are going to be changing the cargo type mid-route. "Abuse" is the wrong word; it's a hacky workaround. 

I understand that I can use refit orders to reinforce a specific cargo type for a given route, which does indeed solve my problem. But it doesn't solve the problem of being unintuitive. I hadn't even considered using refit to solve this problem because in my smooth brain I saw refit orders purely as a means to have vehicles transport multiple types of cargo within the same route, not as the "best-practice" way to enforce a cargo type for single-cargo routes.

Obviously I see now that by having a refit order at the loading station of each route makes it far easier to move vehicles between different routes with different cargos without having to manually reconfigure them, so I'm definitely going to do that. It would just be nice if this method were mentioned in the tutorial, or even if there was a "filtered load" compound order (like "Transfer") that adds both a Refit and Load task. Just something that makes this solution more obvious such that it isn't discovered through community forums or github issue discussions.

**Comment by victorleblais** [comment]
You should be able to create a route/line from a vehicle's order like it currently is.

But I agree that a dedicated route managing window (with the option to create a new route) would be a great addition to the game, similar to the vehicle manager.

As such, this is a duplicate of #21 

Feel free to comment on that issue to add feature you would like.
